src/container/lookup.ts

```diff
@@ -91,7 +91,7 @@ class Lookup<T extends interfaces.Clonable<T>> implements interfaces.Lookup<T> {
         const copy = new Lookup<T>();
 
         this._map.forEach((value, key) => {
-            value.forEach((b) => copy.add(key, b.clone()));
+            value.forEach((b: any) => copy.add(key, b.clone ? b.clone() : b));
```

Comment:
I would create a Typeguard instead:

**isClonable.ts**
```Typescript
   function isClonable<T>(obj: unknown): obj is interfaces.Clonable<T> {
        return (typeof obj === 'object') 
            && (obj !== null)
            && ('clone' in obj)
            && typeof (obj as interfaces.Clonable<T>).clone === 'function';
    }
```

```Typescript
this._map.forEach((value, key) => {
    value.forEach((b: any) => copy.add(key, isClonable<T>(b) ? b.clone() : b));
});
```

In addition, maybe we should not clone a non clonable object but throw an exception instead

Reply:
also agree with the typeguard